src/lightning/data/streaming/dataset.py

```diff
@@ -157,9 +157,12 @@ def __next__(self) -> Any:
         return data
 
 
-def _try_create_cache_dir() -> Optional[str]:
-    if "LIGHTNING_CLUSTER_ID" not in os.environ or "LIGHTNING_CLOUD_PROJECT_ID" not in os.environ:
+def _try_create_cache_dir(input_dir: str) -> Optional[str]:
+    if not os.getenv("LIGHTNING_CLUSTER_ID") or not os.getenv("LIGHTNING_CLOUD_PROJECT_ID"):
```

Comment:
You are changing this because `LIGHTNING_CLUSTER_ID` can be the empty string?

Reply:
reverting this change.